### PR TITLE
Add http methods: head, put, patch, delete

### PR DESCRIPTION
**Pull request by MasterSergius**
Add http methods and tests

### CODE REVIEW

File: src/acachecontrol/acachecontrol.py [hunk 1]
```diff
@@ -27,6 +32,15 @@ def get(self, url, *, allow_redirects=True, **params):
     def post(self, url, *, data, **params):
         return self.request("POST", url, data=data, **params)
 
+    def put(self, url, *, data, **params):
```
Comment (andrii-ivaniuk) [inline src/acachecontrol/acachecontrol.py]: I believe that `put` is not cacheable: https://stackoverflow.com/a/107450/13858551
We need to check other methods (verbs).

File: src/acachecontrol/acachecontrol.py [hunk 2]
```diff
@@ -27,6 +32,15 @@ def get(self, url, *, allow_redirects=True, **params):
...
+        return self.request("PUT", url, data=data, **params)
+
+    def patch(self, url, *, data, **params):
```
Comment (andrii-ivaniuk) [inline src/acachecontrol/acachecontrol.py]: I don't think that we need `*` in methods arguments.
```suggestion
    def patch(self, url, data, **params):
```
It is applicable to other methods.

Comment (MasterSergius) [reply src/acachecontrol/acachecontrol.py]: I used it from original aiohttp as it invokes aiohttp methods. Seems legit.

File: src/acachecontrol/acachecontrol.py (hunk 1 above)
Comment (MasterSergius) [reply src/acachecontrol/acachecontrol.py]: Agree, let me check it and how it is done in CacheControl lib. But, we can set in method self.request any verb, thus this handling should be done in other place.

**Review by andrii-ivaniuk — APPROVED** [review]

**Pull request merged**